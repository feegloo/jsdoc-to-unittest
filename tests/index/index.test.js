const { spawn, spawnJest } = require('jest-helpers/spawn');

describe('#index', () => {
  test('contains.js matches snapshot', async () => {
    expect(
      await spawn('bin/generate', '--input=tests/fixtures/contains.js'),
    ).toMatchSnapshot();
  });

  test('jest passes', async () => {
    expect(
      await spawnJest('--input=tests/fixtures/contains.js'),
    ).not.toBeInstanceOf(Error);
  });

  test('jest passes async test', async () => {
    expect(
      await spawnJest('--input=tests/fixtures/async.js'),
    ).not.toBeInstanceOf(Error);
  });

  test('jest passes get-device-price.js', async () => {
    expect(
      await spawnJest('--input=tests/fixtures/async.js'),
    ).not.toBeInstanceOf(Error);
  });

  test('jest passes array-complement test', async () => {
    expect(
      await spawnJest('--input=tests/fixtures/array-complement.js'),
    ).not.toBeInstanceOf(Error);
  });
});
