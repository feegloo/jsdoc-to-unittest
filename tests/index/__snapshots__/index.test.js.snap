// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#index contains.js matches snapshot 1`] = `
"/* global mock, evaluate */
// Valid tests: 100.00%
export function contains(str1, str2) {
  return str1 && str1.indexOf(str2) !== -1;
}
export default { contains };

/* global mock, evaluate */
// Valid tests: 100.00%

describe('contains', () => {
  test('example 1', () => {
    expect(contains('foo', 'fooo')).toBe(false);
  });

  test('example 2', () => {
    expect(contains('hey', 'ey')).toBe(true);
  });
});
"
`;
